xbmc/guilib/GUIInfoTypes.h

```diff
@@ -125,6 +126,27 @@ class CGUIInfoLabel
    */
   static std::string ReplaceAddonStrings(const std::string &label);
 
+  typedef boost::function<std::string(const std::string&)> StringReplacerFunc;
+
+  /*!
+   \brief Replaces instances of $strKeyword[value] with the appropriate resolved string
+   \param strInput text to replace
+   \param strKeyword keyword to look for
+   \param func function that does the actual replacement of each bracketed value found
+   \param strOutput the output string
+   \return whether anything has been replaced.
+   */
+  static bool ReplaceDollarKeywordStrings(const std::string &strInput, const std::string &strKeyword, const StringReplacerFunc &func, std::string &strOutput);
```

Comment:
> Parameter references are not resolved in a "global" context and are not really "info" labels.

Agree and makes sense, leaven everything as it is. 

> Btw, if you check out early posts on the discussion page, you'll see that I originally wanted to put ReplaceDollarKeywordStrings() into some kind of a util class, but simply didn't find the appropriate one.

For now we can leave it in CGUIInfoTypes, but I agree that this should be moved elsewhere.

> Perhaps we can just think of a better (but descriptive) name for ReplaceDollarKeywordStrings()? I am open to suggestions.

That would be much appreciated. IMO we talk about special keywords so what do you think about name it `ReplaceSpecialKeywordStrings()` .. sure a small detail ;)